### PR TITLE
Feature/view

### PR DESCRIPTION
**Pull request by Norskan**
Please do a review! thanks

### CODE REVIEW

File: root/application/.gitignore [hunk 1]
```diff
@@ -1 +1,2 @@
 /target/
+/target/
```
Comment (ghost) [inline root/application/.gitignore]: Better safe than sorry :wink:

File: root/communication-rmi/.gitignore [hunk 2]
```diff
@@ -1 +1,2 @@
 /target/
+/target/
```
Comment (ghost) [inline root/communication-rmi/.gitignore]: Better safe than sorry :wink:

File: root/view/src/main/java/at/fhv/itb5c/css/BaseStile.css [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+.label-header {
+    -fx-font-size: 32pt;
+    -fx-font-family: "Arial";
+    -fx-text-fill: black;
+}
+
+.button {
+	-fx-font-size: 9pt;
+	-fx-font-family: "Arial";
+    -fx-text-fill: black;
```
Comment (ghost) [inline root/view/src/main/java/at/fhv/itb5c/css/BaseStile.css]: tabs or spaces?

File: root/view/src/main/java/at/fhv/itb5c/model/UserModel.java [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+	private StringProperty _adress;
+	private StringProperty _eMail;
+	private StringProperty _telephonenumber;
+	private Gender _gender;
+	private ObjectProperty<LocalDate> _birthDate;
+	private ObservableList<TypeOfSport> _typeOfSports;
+	private DoubleProperty _membershipFee;
+	private UserRole _userRole;
+
+	public UserModel() {
```
Comment (ghost) [inline root/view/src/main/java/at/fhv/itb5c/model/UserModel.java:34]: add a constructor that takes a `UserDTO` as first parameter

File: root/view/src/main/java/at/fhv/itb5c/util/RouteProvider.java [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+		if(_instance == null) {
+			_instance = new RouteProvider();
+		}
+		
+		return _instance;
+	}
+	
+	public void addRoot(Object controller, String route) throws ControllerHasRouteException {
+		if(_rootMapping.containsKey(controller)) {
+			throw new ControllerHasRouteException("The controller " + controller.getClass() + "has a registrated route.");
```
Comment (ghost) [inline root/view/src/main/java/at/fhv/itb5c/util/RouteProvider.java]: just overwrite it?

File: root/view/src/main/java/at/fhv/itb5c/view/login/LoginController.java [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+public class LoginController {
+	
+	@FXML
+	BorderPane _rootPane;
+	
+	@FXML
+	private Button _loginButton;
+
+	@FXML
+	public void LoginButton_OnMouseReleased_Button_Handler(MouseEvent event) {
```
Comment (ghost) [inline root/view/src/main/java/at/fhv/itb5c/view/login/LoginController.java]: camelCase

File: root/view/src/main/java/at/fhv/itb5c/view/mainview/MainViewController.java [hunk 7]
```diff
@@ -0,0 +1,86 @@
...
+	BorderPane _rootPane;
+	
+	@FXML
+	Pane _mainPanel;
+	
+	@FXML
+	private Button _loginButton;
+
+	@FXML
+	public void LoginMenueItemActionHandler(ActionEvent event) {
```
Comment (ghost) [inline root/view/src/main/java/at/fhv/itb5c/view/mainview/MainViewController.java]: camelCase

File: root/view/src/main/java/at/fhv/itb5c/view/mainview/MainViewController.java [hunk 8]
```diff
@@ -0,0 +1,86 @@
...
+		} catch (IOException e) {
+			// TODO add logging
+			e.printStackTrace();
+		}
+		
+		loginStage.show();
+	}
+	
+	@FXML
+	public void CloseMenueItemActionHandler(ActionEvent event) {
```
Comment (ghost) [inline root/view/src/main/java/at/fhv/itb5c/view/mainview/MainViewController.java]: camelCase

File: root/view/src/main/java/at/fhv/itb5c/view/mainview/MainViewController.java [hunk 9]
```diff
@@ -0,0 +1,86 @@
...
+		// TODO close resources etc . . .
+		((Stage)_rootPane.getScene().getWindow()).close(); 
+	}
+	
+	@FXML
+	public void CreateUserMenueItemActionHandler(ActionEvent event) {
```
Comment (ghost) [inline root/view/src/main/java/at/fhv/itb5c/view/mainview/MainViewController.java]: camelCase

File: root/view/src/main/java/at/fhv/itb5c/view/user/UserView.fxml [hunk 10]
```diff
@@ -0,0 +1,75 @@
...
+                <ColumnConstraints hgrow="SOMETIMES" minWidth="10.0" />
+              </columnConstraints>
+              <rowConstraints>
+                <RowConstraints minHeight="10.0" prefHeight="30.0" vgrow="SOMETIMES" />
+              </rowConstraints>
+               <children>
+                  <Button minWidth="-Infinity" mnemonicParsing="false" onMouseReleased="#saveButtonMouseReleasedEventHandler" text="Save" />
+                  <Button minWidth="-Infinity" mnemonicParsing="false" onMouseReleased="#cancleButtonMouseReleasedEventHandler" text="Cancel" GridPane.columnIndex="1" />
+               </children>
+            </GridPane> -->
```
Comment (ghost) [inline root/view/src/main/java/at/fhv/itb5c/view/user/UserView.fxml]: do we need this comment?

File: root/view/src/main/java/at/fhv/itb5c/view/user/UserViewController.java [hunk 11]
```diff
@@ -0,0 +1,215 @@
...
+		detailState ("detailState"),
+		modifieState ("modifieState");
+		
+		private final String _name;  
+		
+		private ViewState(String name) {
+		        _name = name;
+		    }
+
+	    public String toString() {
```
Comment (ghost) [inline root/view/src/main/java/at/fhv/itb5c/view/user/UserViewController.java]: tabs and spaces..

File: root/view/src/main/java/at/fhv/itb5c/view/user/UserViewController.java [hunk 12]
```diff
@@ -0,0 +1,215 @@
...
+		else {
+			return false;
+		}
+	}
+
+	public void close() {
+		_panelCloseHandler.close();
+	}
+	
+	private HashMap<ViewState, UserViewState> _userViewStates;
```
Comment (ghost) [inline root/view/src/main/java/at/fhv/itb5c/view/user/UserViewController.java]: Use the interface [`Map`](https://docs.oracle.com/javase/8/docs/api/java/util/Map.html)

File: root/view/src/main/java/at/fhv/itb5c/view/user/states/DetailUserViewControlls.java [hunk 13]
```diff
@@ -0,0 +1,43 @@
...
+	}
+
+	@Override
+	public void activate() {
+		_userViewController.setDisable(true);
+	}
+	
+	@FXML 
+	public void modifyButtonMouseReleasedEventHandler(MouseEvent mouseEvent) {
+		//TODO: switch to modifie view
```
Comment (ghost) [inline root/view/src/main/java/at/fhv/itb5c/view/user/states/DetailUserViewControlls.java:36]: ??

File: root/view/src/test/java/at/fhv/itb5/view/AppTest.java [hunk 14]
```diff
@@ -0,0 +1,38 @@
...
+import junit.framework.Test;
+import junit.framework.TestCase;
+import junit.framework.TestSuite;
+
+/**
+ * Unit test for simple App.
+ */
+public class AppTest 
+    extends TestCase
+{
```
Comment (ghost) [inline root/view/src/test/java/at/fhv/itb5/view/AppTest.java]: we're not in C# :wink:

File: root/view/src/main/java/at/fhv/itb5c/model/UserModel.java (hunk 4 above)
Comment (Norskan) [inline root/view/src/main/java/at/fhv/itb5c/model/UserModel.java:34]: will be added

File: root/view/src/main/java/at/fhv/itb5c/view/user/states/DetailUserViewControlls.java (hunk 13 above)
Comment (Norskan) [inline root/view/src/main/java/at/fhv/itb5c/view/user/states/DetailUserViewControlls.java:36]: just a stub where logic is added when we implement the modifie user story

**Comment by ghost** [comment]
`ack`

**Pull request merged**